src/persist-client/src/cfg.rs

```diff
@@ -258,12 +267,67 @@ pub struct DynamicConfig {
     consensus_connect_timeout: RwLock<Duration>,
     sink_minimum_batch_updates: AtomicUsize,
 
+    next_listen_batch_retryer: RwLock<RetryParameters>,
+
     // TODO: Figure out how to make these dynamic.
     compaction_minimum_timeout: Duration,
     consensus_connection_pool_ttl: Duration,
     consensus_connection_pool_ttl_stagger: Duration,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Arbitrary, Serialize, Deserialize)]
+pub struct RetryParameters {
+    pub initial_backoff: Duration,
+    pub multiplier: u32,
+    pub clamp: Duration,
+}
+
+impl RetryParameters {
+    pub(crate) fn into_retry(self, now: SystemTime) -> Retry {
+        let seed = now
+            .duration_since(UNIX_EPOCH)
+            .map_or(0, |x| u64::from(x.subsec_nanos()));
+        Retry {
+            initial_backoff: self.initial_backoff,
+            multiplier: self.multiplier,
+            clamp_backoff: self.clamp,
+            seed,
+        }
+    }
+}
+
+impl RustType<ProtoRetryParameters> for RetryParameters {
+    fn into_proto(&self) -> ProtoRetryParameters {
+        ProtoRetryParameters {
+            initial_backoff: Some(self.initial_backoff.into_proto()),
+            multiplier: self.multiplier,
+            clamp: Some(self.clamp.into_proto()),
+        }
+    }
+
+    fn from_proto(proto: ProtoRetryParameters) -> Result<Self, TryFromProtoError> {
+        let initial_backoff: Option<Duration> = proto.initial_backoff.into_rust()?;
+        let clamp: Option<Duration> = proto.clamp.into_rust()?;
+
+        let Some(initial_backoff) = initial_backoff else {
```

Comment:
`into_rust_if_some`?

Reply:
THANK YOU I knew there had to be a better way